x/mongo/driver/topology/connection_test.go

```diff
@@ -110,6 +110,17 @@ func TestConnection(t *testing.T) {
 				assert.NotNil(t, err, "expected connect error %v, got nil", want)
 				assert.Equal(t, want, got, "expected error %v, got %v", want, got)
 			})
+			t.Run("cancelConnectContext is nil after connect", func(t *testing.T) {
+				conn, err := newConnection(context.Background(), address.Address(""))
+				assert.Nil(t, err, "newConnection shouldn't error. got %v; want nil", err)
+
+				go func() {
```

Comment:
Just realized we're not actually waiting for the goroutine to finish, so it's possible that the test finishes before the goroutine even starts running. We should use a `sync.WaitGroup` to ensure it's done. I'd recommend looking at https://golang.org/pkg/sync/#WaitGroup and the linked example to get an idea of what you should do.

Reply:
done